src/pseudopeople/noise_functions.py

```diff
@@ -45,6 +42,86 @@ def omit_rows(
     return noised_data
 
 
+def _get_census_do_not_respond_demographic_probabilities(
+    population: pd.DataFrame,
+    base_probability: float = data_values.DO_NOT_RESPOND_BASE_PROBABILITY,
+) -> pd.Series:
+    """
+    Helper function for do_not_respond noising based on demography of age, race/ethnicity, and sex.
+
+    :param population: a dataset containing records of simulants
+    :param base_probability: base probability for do_not_respond
+    :return: a pd.Series of probabilities
+    """
+    probabilities = pd.Series(base_probability, index=population.index)
+    probabilities += (
+        population["race_ethnicity"]
+        .astype(str)
+        .map(data_values.DO_NOT_RESPOND_ADDITIVE_PROBABILITY_BY_RACE)
+    )
+    for s in ["Female", "Male"]:
+        sex_mask = population["sex"] == s
+        age_lookup = (
+            data_values.DO_NOT_RESPOND_ADDITIVE_PROBABILITY_BY_SEX_AGE[
+                data_values.DO_NOT_RESPOND_ADDITIVE_PROBABILITY_BY_SEX_AGE["sex"] == s
+            ]
+            .drop(["sex"], axis=1)
+            .rename(columns={"age_interval": "age"})
+        )
+
+        age_bins = pd.cut(
+            x=population[sex_mask]["age"],
+            bins=pd.IntervalIndex(age_lookup["age"]),
+        )
+        probabilities[sex_mask] += age_bins.map(age_lookup.set_index("age").squeeze()).astype(
+            float
+        )
+    return probabilities
+
+
+def omit_target_rows(
+    dataset_name: str,
+    dataset_data: pd.DataFrame,
+    configuration: ConfigTree,
+    randomness_stream: RandomnessStream,
+) -> pd.DataFrame:
+    """
+    Applies targeted omission based on demographic model for census and surveys.
+
+    :param dataset_name: Dataset object name being noised
+    :param dataset_data:  pd.DataFrame of one of the form types used in Pseudopeople
+    :param configuration: ConfigTree object containing noise level values
+    :param randomness_stream: RandomnessStream object to make random selection for noise
+    :return: pd.DataFrame with rows from the original dataframe removed
+    """
+    supported_datasets = [DatasetNames.ACS, DatasetNames.CPS, DatasetNames.CENSUS]
+    if dataset_name not in supported_datasets:
+        raise ValueError(
+            f"Unexpected dataset_name {dataset_name}. Valid values are: {supported_datasets}"
+        )
+
+    noise_levels = _get_census_do_not_respond_demographic_probabilities(dataset_data)
```

Comment:
Can we have a clearer name for this helper function like `get_census_omission_noise_levels`?